data/processor.py

```diff
@@ -580,6 +596,37 @@ def import_opendata_sections():
     print 'Added %s sections and updated %s sections' % (
             num_added, num_updated)
 
+def import_scholarships():
+    num_added = 0
+    num_updated = 0
+
+    filenames = glob.glob(os.path.join(os.path.dirname(__file__),
+            c.SCHOLARSHIPS_DATA_DIR, '*.json'))
+
+    for filename in filenames:
+        with open(filename, 'r') as f:
+            data = json.load(f).get('data')
+
+            for scholarship_data in data:
+                scholarship_dict = _clean_scholarship(scholarship_data)
+
+                existing_scholarship = m.Scholarship.objects(
+                        id=scholarship_dict['id']
+                ).first()
+
+                if existing_scholarship:
+                    for key, val in scholarship_dict.iteritems():
+                        if key != 'id':
+                            existing_scholarship[key] = val
+                    existing_scholarship.save()
+                    num_updated += 1
+                else:
+                    m.Scholarship(**scholarship_dict).save()
+                    num_added += 1
```

Comment:
Nice upsert pattern (I guess this was taken from somewhere, but still, good of you to use it). Though it'd be ideal if MongoEngine or Mongo actually gave us a real upsert function.